directives/animate/animate.js

```diff
@@ -291,50 +289,32 @@ class AnimationState {
 }
 
 export class Hide extends AsyncDirective {
-	state;
-	constructor(propertyPart) {
-		super(propertyPart);
-		if (!(propertyPart.type === PropertyPart) || propertyPart.committer.name !== 'animate') {
-			throw new Error('animation directives must be used with "animate" property');
-		}
+	constructor(partInfo) {
+		super(partInfo);
+		this.state = new AnimationState(partInfo);
 	}
-	// Do SSR-compatible rendering (arguments are passed from call site)
-	render(part) {
-		// Previous state available on class field
-		if (this.state === undefined) {
-			this.state = new AnimationState(part);
-		}
+	render() {
 		return noChange;
 	}
-	update(part, opts) {
-		/* Any imperative updates to DOM/parts would go here */
+	update(part, [opts]) {
 		opts = opts || {};
 		this.state.hide(opts);
 		return this.render(part);
```

Comment:
Wondering why this one didn't need to be changed to opts as well? I would think the Hide and Show would be identical with the exception of this.state.hide/show function calls.

Reply:
Sorry that was just a mistake on my "part" haha. Render should always take `opts` and `update()` gets passed both the part and the parameters. In our case though `render()` doesn't do anything at all, so that's why this isn't blowing up.